Make the rollup/sync buttons more distinct, and point to sync docs

jieyouxu opened the pull request:
Currently, it's not obvious that "synchronize" is a dangerous operation, and there's no indication for the contributors how to properly perform a sync. Make this more clear by making the buttons larger and color-coded, and include a backlink to sync procedure.

### Preview

![Screenshot 2025-05-09 235919](https://github.com/user-attachments/assets/2cd73859-335a-45a8-b2cc-f7e5ffbcba2b)

r? @pietroalbini (as you said you'd like to review such a change)

Zalathar commented:
This has made the bors queue page a lot harder for me to use.

I primarily want to *look* at the queue, and doing that is so much harder now that I have gigantic buttons constantly yelling at me, especially the synchronize button that I basically never ever want to press.

And it's non-trivial to write a user stylesheet that undoes these changes.

jieyouxu commented:
> This has made the bors queue page a lot harder for me to use.
> 
> I primarily want to *look* at the queue, and doing that is so much harder now that I have gigantic buttons constantly yelling at me, especially the synchronize button that I basically never ever want to press.
> 
> And it's non-trivial to write a user stylesheet that undoes these changes.

If instead of changing the button, the sync button becomes more like the rollup button that has another confirmation step, would that be less intrusive?

Zalathar commented:
> If instead of changing the button, the sync button becomes more like the rollup button that has another confirmation step, would that be less intrusive?

Yeah, I think that would be fine.

Another option might be to move the sync button to the *bottom* of the page, so that it isn't disruptive (or tempting) in the normal scenario of looking at the top of the page, which is where the key information is.

pietroalbini commented:
I agree with moving the synchronize button at the bottom of the page, and maybe reverting the styling of the "create rollup" button the same as it was before. We should still make the synchronize button be styled in a scary way.

jieyouxu commented:
> Another option might be to move the sync button to the _bottom_ of the page, so that it isn't disruptive (or tempting) in the normal scenario of looking at the top of the page, which is where the key information is.

Ok, reverted rollup style changes, and moved sync button to bottom of page but still keep it scary: https://github.com/rust-lang/homu/pull/232
